distbench_engine.cc

```diff
@@ -542,7 +550,13 @@ absl::Status DistBenchEngine::RunTraffic(const RunTrafficRequest* request) {
 
 void DistBenchEngine::CancelTraffic() {
   LOG(INFO) << engine_name_ << ": Got CancelTraffic";
-  canceled_.Notify();
+  if (!canceled_.HasBeenNotified()) {
+    canceled_mu_.Lock();
```

Comment:
style: the lock can be freed using a absl::MutexLock now that there are convenient braces nearby

Reply:
Done